test/testdrive/kafka-avro-sinks-doc-comments.td

```diff
@@ -0,0 +1,97 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test Avro UPSERT sinks doc comments
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+ALTER SYSTEM SET enable_comment = true;
```

Comment:
> This does not prevent the user from explicitly adding a DOC ON options in their create sink sql though.

We should add a separate feature flag for the use of `DOC ON` comments, so that we can properly follow the feature lifecycle for this feature. It should start out in private preview while we're validating its design and implementation with Pluralsight. That way we can make breaking changes if necessary.

Reply:
Sounds good. Let me add that.

Reply:
Done! Put it behind `enable_sink_doc_on_option`